inbox/mailsync/backends/imap/common.py

```diff
@@ -156,10 +156,14 @@ def update_metadata(account_id, folder_id, folder_role, new_flags, session):
 
     account = Account.get(account_id, session)
     change_count = 0
-    for item in session.query(ImapUid).filter(
-        ImapUid.account_id == account_id,
-        ImapUid.msg_uid.in_(new_flags),
-        ImapUid.folder_id == folder_id,
+    for item in (
+        session.query(ImapUid)
+        .filter(
+            ImapUid.account_id == account_id,
+            ImapUid.folder_id == folder_id,
+            ImapUid.msg_uid.in_(new_flags),
+        )
+        .with_hint("FORCE INDEX (ix_imapuid_account_id_folder_id_msg_uid_desc)")
```

Comment:
Here this is not covering index, because we retrieve other columns, but it does not make things slower, it just switches from the old to the new index.